docs/dev/release-process.md

```diff
@@ -28,5 +28,27 @@ While the `[do not deploy]` can be anywhere in the title, it is recommended to p
 The deployment pipeline will analyse the message of the merge commit (which will be contain the PR title) and make a decision based on it.
 
 ## Releasing to production
-The process to promote a build from `stage` to `production` is currently manually initiated by SRE.
-[This ticket](https://mozilla-hub.atlassian.net/browse/CONSVC-1566) (requires LDAP) deals with automating the process.
+Developers with write access to the Merino repository can initiate a deployment to production after a Pull-Request on the Merino GitHub repository is merged to the `main` branch. This can be done by:
+
+1. opening the [circleci dashboard][circleci_dashboard];
+2. looking up the pipeline named `merino <PR NUMBER>` running in the `main-workflow`; this pipeline should either be in a running status (if the required test jobs are still running) or in the "on hold" status, with the `unhold-to-deploy-to-prod` being held;
+3. once in the "on hold" status, with all the other jobs successfully completed, clicking on the "thumbs up" action on the `unhold-to-deploy-to-prod` job row will approve it and trigger the deployment, unblocking the `deploy-to-prod` job;
+4. developers **must** monitor the [Merino Application & Infrastructure][merino_app_info] dashboard for any anomaly, for example significant changes in HTTP response codes, increase in latency, cpu/memory usage (most things under the infrastructure heading).
+
+[circleci_dashboard]: https://app.circleci.com/pipelines/github/mozilla-services/merino?branch=main&filter=all
+[merino_app_info]: https://earthangel-b40313e5.influxcloud.net/d/Cm83vS57z/merino-application-and-infrastructure?orgId=1&refresh=1m
+
+## What to do if production is broken?
+Don't panic and follow the instructions below:
+
+- depending on the severity of the problem, decide if this warrants [kicking off an incident][incident_docs];
+- if the root cause of the problem can be identified in a relatively small time, create a PR for the fix.
+    - verify the fix locally;
+    - verify the fix on `stage`, after it is reviewed by a Merino developer and merged;
+    - [deploy it to production](#releasing-to-production).
+
+**OR**
+
+- if the root cause of the problem is harder to track down, revert the last commit and then [deploy the revert commit to production](#releasing-to-production).
```

Comment:
Another clarification question: what if the problematic commit is not the last commit and some subsequent commits after it are also deployed in production. I think we'd want to provide developers with a way to easily revert to the last working commit. This can be tricky though with a highly active repo, but still worth thinking about.

Reply:
This is a great question, and I believe that this question boils down to the philosophical difference between Continuous Deployment and Continuous Delivery: with Continuous Delivery, you delivery whenever you feel like, maybe weekly, maybe not, but you potentially put together multiple changesets.

With Continuous Deployment, the idea is to deploy every time you merge to main. That's the part that reduces the risk of breaking, because when releasing on each single merge you can easily tell if you're merging something problematic. Ideally, you don't uncover with a deployment from today something that broke 15 days ago. Because the 15 days ago thing would have triggered problems on that deployment. If the interaction with the new thing is the problem, then you still solve it by reverting/disabling the newest change.

The need to go back in time to deploy an old commit, I believe, sort of defeats the point of Continuous Deployment. It's still doable, manually, by SRE though (which is the current state of things anyway).